libs/platform/src/lib/table/table.service.ts

```diff
@@ -226,9 +263,126 @@ export class TableService {
     recalculateColumnsWidth(): void {
         this._tableColumnsWidth$.next();
     }
+
+    /** Register focusable table cell to have ability navigate over it with arrow buttons */
+    registerFocusableTableCell(id: string, tableCell: FdpCellSelectableDirective): void {
+        this._tableCellsMap.set(id, tableCell);
+    }
+
+    /** Register focusable table cell to have ability navigate over it with arrow buttons */
+    clearFocusableTableCells(): void {
+        this._tableCellsMap.clear();
+    }
+
+    /** Navigate over registered focusable table cells */
+    focusNextTableCell(currCellNavId: TableCellNavigationId, event?: KeyboardEvent): void {
+        if (!event) {
+            const currCell = this._tableCellsMap.get(currCellNavId);
+
+            if (currCell) {
+                currCell.focus();
+            }
+
+            return;
+        }
+
+        const [rowIndex, colIndex] = currCellNavId.split(',');
+        let nextCellId: string;
+
+        if (KeyUtil.isKeyCode(event, UP_ARROW)) {
+            nextCellId = `${+rowIndex - 1},${colIndex}`;
+        }
+
+        if (KeyUtil.isKeyCode(event, DOWN_ARROW)) {
+            nextCellId = `${+rowIndex + 1},${colIndex}`;
+        }
+
+        if (KeyUtil.isKeyCode(event, LEFT_ARROW)) {
+            nextCellId = `${rowIndex},${+colIndex - (this._rtl ? -1 : 1)}`;
+        }
+
+        if (KeyUtil.isKeyCode(event, RIGHT_ARROW)) {
+            nextCellId = `${rowIndex},${+colIndex + (this._rtl ? -1 : 1)}`;
+        }
+
+        const nextCell = this._tableCellsMap.get(nextCellId);
+        if (nextCell) {
+            this._tableCellsMap.get(nextCellId).focus();
+        }
+    }
+
+    /** Set currently focused cell */
+    setFocusedTableCell(cellId): void {
+        this._focusedTableCellNavId = cellId;
+    }
+
+    /** Reset currently focused cell to initial value */
+    resetFocusedTableCell(): void {
+        this._focusedTableCellNavId = FIRST_CELL_NAVIGATION_ID;
+    }
+
+    /** Mark that focus inside the cell */
+    setFocusInsideCell(): void {
+        this._isFocusInsideTableCell = true;
+    }
+
+    /** Mark that focus isn't inside the cell */
+    removeFocusInsideCell(): void {
+        this._isFocusInsideTableCell = false;
+    }
+
+    /** Focus on table cell inner any first focusable element */
+    processFocusInsideCell(tableCellNavId: TableCellNavigationId): void {
+        const tableCell = this._tableCellsMap.get(tableCellNavId);
+
+        if (this.isFocusInsideTableCell) {
+            tableCell.focus();
+            this.removeFocusInsideCell();
+            return;
+        }
+
+        const focusableElements = getFocusableElements(tableCell._elRef.nativeElement);
+
+        if (focusableElements.length) {
+            focusableElements[0].focus();
+            this.setFocusInsideCell();
+        }
```

Comment:
I think in this case you can have an extra param in `getFocusableElements` to run `querySelector` instead of `querySelectorAll`.

Reply:
Because you need only first entry. `querySelector` is much faster than `querySelectorAll`. 
https://measurethat.net/Benchmarks/Show/5353/0/queryselector-vs-queryselectorall-vs-getelementsbyclass
![image](https://user-images.githubusercontent.com/33101123/145365983-6435f08f-b8b4-4821-8fe3-4c56d5749959.png)